doc/getting_started.rst

```diff
@@ -14,155 +14,186 @@ these usually are:
 
 .. code-block:: bash
 
-   $ apt-get install python3 python3-virtualenv python3-pip virtualenv
+   apt-get install python3 python3-virtualenv python3-pip virtualenv
 
 
 In many cases, the easiest way is to install labgrid into a virtualenv:
 
 .. code-block:: bash
 
-    $ virtualenv -p python3 labgrid-venv
-    $ source labgrid-venv/bin/activate
+    virtualenv -p python3 labgrid-venv
+    source labgrid-venv/bin/activate
 
 Start installing labgrid by cloning the repository and installing the
 requirements from the `requirements.txt` file:
 
 .. code-block:: bash
 
-    $ git clone https://github.com/labgrid-project/labgrid
-    $ cd labgrid && pip install -r requirements.txt
-    $ python3 setup.py install
-
-.. note::
-   Previous documentation recommended the installation as via pip (`pip3 install
-   labgrid`).
-   This lead to broken installations due to unexpected incompatibilities with
-   new releases of the dependencies.
-   Consequently we now recommend using pinned versions from the
-   `requirements.txt` file for most use cases.
-
-   labgrid also supports the installation as a library via pip, but we only
-   test against library versions specified in the requirements.txt file.
-   Thus when installing directly from pip you have to test compatibility
-   yourself.
-
-.. note::
-   If you are installing via pip and intend to use Serial over IP (RFC2217),
-   it is highly recommended to uninstall pyserial after installation and replace
-   it with the pyserial version from the labgrid project:
-
-      .. code-block:: bash
-
-          $ pip uninstall pyserial
-          $ pip install https://github.com/labgrid-project/pyserial/archive/v3.4.0.1.zip#egg=pyserial
-
-   This pyserial version has two fixes for an Issue we found with Serial over IP
-   multiplexers. Additionally it reduces the Serial over IP traffic considerably
-   since the port is not reconfigured when labgrid changes the timeout (which is
-   done inside the library a lot).
+    git clone https://github.com/labgrid-project/labgrid
+    cd labgrid && pip install -r requirements.txt
+    python3 setup.py install
```

Comment:
pip actually supports installing from a git repository, right away:
```sh
pip install git+https://github.com/labgrid-project/labgrid/@master#egg=labgrid
```
However, I think that `master` should likely replaced with Sphinx' [`release`](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-release) variable. Which, in turn, should be set to the git ref (`v0.2.0`, `master`, ...) .

Reply:
@leiflm I thought it wouldn't use the requirements.txt from the repository in that case (which is the main reason we currently suggest to do it this way).

Reply:
> I thougt it wouldn't use the `requirements.txt` from the repository in that case.

@jluebbe you are, most likely, correct. Why wouldn't you change the requirements in `setup.py` accordingly, though?

Anyway, pip also supports to specify the `requirements.txt` using a URI:
```sh
pip install -r https://raw.githubusercontent.com/labgrid-project/labgrid/master/requirements.txt
```

Reply:
The Python packaging User Guide does not recommend pinning dependencies in `setup.py`, see [here](https://packaging.python.org/discussions/install-requires-vs-requirements/).

Reply:
Thanks for the insight on why `setup.py` is not adjusted. I didn't see beforehand, that the current `requirements.txt` actually pinned all versions of the required packages. Reading the docs, I assumed it would just be a bit more restrictive about package versions used (i.e. `A>=1,<2`).

I assume that nobody has a setup at hand to verify which packages actually need to be kept pinned down? I guess that would be an issue of its own :)